.travis.yml

```diff
@@ -8,8 +8,8 @@ env:
   global:
     # TODO: -Werror
     # TODOL -fsanitize=address -fsanitize=undefined
-    - CFLAGS: "-Wall -Wextra"
-    - CXXFLAGS: "-Wall -Wextra"
+    - CFLAGS: "-Wall -Wextra -Werror"
```

Comment:
seams to be ignored by the build scripts since I can see warnings here: https://travis-ci.org/pmix/pmix/builds/521290397

Reply:
We shouldn't be ignoring it - I'll have to look. However, we cannot do anything about the warning you are seeing there as it is an artifact of flex, not us. We've had to endure it since the beginning. IIRC, somebody (@ggouaillardet ? ) came up with a way of silencing it once - but I'm not sure what might have brought it back, assuming my recollection is correct.

Reply:
The "best" solution to the flex warning that I have seen is to not pass the `-Werror` to that specific part of the build. Not great, but it's something.

Reply:
@rhc54 did you find a way to pass my CFLAGS to the build process? I am trying to add address sanitizers to your CI so OpenMPI applications can be used with less suppression files in the future.

Reply:
I did - unfortunately, it causes the build to fail in many places as the picky mode generates warnings over the slightest thing (e.g., a function with a parameter that doesn't get used in that specific case because we have multiple variations of that function in our plugin system). I'm having to work thru the code base to get things to co-exist with it. Can't make it a high priority right now, so this may take a while to complete. Will try to periodically post progress on it.

Reply:
Ok, but can you tell my why exporting `CFLAGS` (as in the CI script) does not influence the build? How did you trigger `-Werror` to be properly applied?

Reply:
Urgh, autotools does not play well with `-Werror` either: https://lists.gnu.org/archive/html/bug-autoconf/2017-01/msg00008.html